internal/security/secretstore/password.go

```diff
@@ -0,0 +1,216 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package secretstore
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/cloudflare/gokey"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+)
+
+// CredentialGenerator returns a credential generated with random algorithm for secret store
+type CredentialGenerator interface {
+	Generate(string) (string, error)
+}
+
+// GokeyGenerator implements the CredentialGenerator interface using the gokey library
+// using tokenPath as the gokey master password and accepting the realm as the argument
+// to the Generate method
+type GokeyGenerator struct {
+	tokenPath string
+}
+
+func NewGokeyGenerator(tokenPath string) *GokeyGenerator {
+	return &GokeyGenerator{tokenPath: tokenPath}
+}
+
+func (gk GokeyGenerator) Generate(realm string) (string, error) {
+	passSpec := gokey.PasswordSpec{
+		Length:         16,
+		Upper:          3,
+		Lower:          3,
+		Digits:         2,
+		Special:        1,
+		AllowedSpecial: "",
+	}
+	t, err := GetAccessToken(gk.tokenPath)
+	if err != nil {
+		return "", err
+	}
+	return gokey.GetPass(t, realm, nil, &passSpec)
+}
+
+type CredCollect struct {
+	Pair UserPasswordPair `json:"data"`
+}
+
+type UserPasswordPair struct {
+	User     string `json:"user,omitempty"`
+	Password string `json:"password,omitempty"`
+}
+
+type Cred struct {
+	client    internal.HttpCaller
+	tokenPath string
+	g         CredentialGenerator
```

Comment:
Can we spell out this field name in struct like `generator`?  I am a fan of self-document code so full name is more descriptive and clear to the developer.